scripts/initramfs/init.x86

```diff
@@ -306,38 +304,41 @@ if [ ! -z "${BOOTDELAY}" ]; then
   sleep ${BOOTDELAY}
 fi
 
-print_msg "Detecting additional removable devices"
+print_msg "Checking for a volumio rootfs update on a USB device"
 mkdir /mnt/usb
-usb_done=/dev/
-for devlink in /dev/sd*; do
-  # only checking the first partition of an additional device!
-  usb_dev=$(echo ${devlink} | sed 's/[0-9]*//g')
-  if [ ${usb_done} != ${usb_dev} ] && [ ${BOOT_DEVICE} != ${usb_dev} ]; then
-    mount -t auto $(echo ${usb_dev}1) /mnt/usb >/dev/null 2>&1
-    print_msg "Checking " $usb_dev"1 for a volumio rootfs update"
+for devlink in /dev/sd[a-z]; do
+  # do the first partition of a usb device, must be removable and not the boot device!
+  dev_current=$(echo ${devlink} | sed 's/[0-9]*//g')
+  if [ $(lsblk ${dev_current} --list -no tran) == usb ] && \
+     [ ${BOOT_DEVICE} != ${dev_current} ]; then
+    mount -t vfat ${dev_current}1 /mnt/usb >/dev/null 2>&1
+    print_msg "Checking ${dev_current}1 for a volumio rootfs update"
     if [ -e /mnt/usb/*.fir ]; then
       print_msg "New rootfs firmware found"
-      print_msg "Updating will take several minutes, please wait...."
+      print_msg "Updating can take several minutes, please wait...."
+      mkdir /mnt/boot
+      mount -t vfat "${BOOTPART}" /mnt/boot
       volumio-init-updater
-      sync
       print_msg "USB Update applied"
       umount /mnt/usb >/dev/null 2>&1
       rm -r /mnt/boot
+      rm -r /mnt/usb
       print_msg "Restarting"
       echo b >/proc/sysrq-trigger
       break
     fi
     umount /mnt/usb >/dev/null 2>&1
   fi
-  usb_done=${usb_dev}
 done
-print_msg "Checking for factory reset on an additional device"
-for devlink in /dev/sd*; do
-  # only checking the first partition of an additional device!
-  usb_dev=$(echo ${devlink} | sed 's/[0-9]*//g')
-  if [ ${usb_done} != ${usb_dev} ] && [ ${BOOT_DEVICE} != ${usb_dev} ]; then
-    mount -t auto $(echo ${usb_dev}1) /mnt/usb >/dev/null 2>&1
-    print_msg "Checking " $usb_dev"1 for a factory reset"
+
+print_msg "Checking for factory reset on a USB device"
+for devlink in /dev/sd[a-z]; do
+  # do the first partition of a usb device, must be removable and not the boot device!
+  dev_current=$(echo ${devlink} | sed 's/[0-9]*//g')
```

Comment:
Same as above..

Reply:
Well spotted, missed one when reducing the code
This will be fixed with https://github.com/volumio/Build/pull/492/commits/7c58cd5896b4e09dfdab04ad700d648342fa7821